packages/types/src/project.ts

```diff
@@ -136,6 +136,7 @@ export interface SubqlCustomDatasource<
   assets: Map<string, CustomDataSourceAsset>;
   processor: FileReference;
   abi?: string; // Should be a key of assets
+  address?: string;
```

Comment:
for `abi` and `address` here, they are defined by CustomProcessor not aware by us.
i suggest 
```
export interface SubqlCustomDatasource<
  K extends string = string,
  T extends SubqlNetworkFilter = SubqlNetworkFilter,
  M extends SubqlMapping = SubqlMapping<SubqlCustomHandler>
> extends ISubqlDatasource<M, T> {
  kind: K;
  assets: Map<string, CustomDataSourceAsset>;
  processor: FileReference;
  [more: string]: unknown;
}
```

Reply:
Its not possible to do this with class validator

Reply:
because of `forbidNonWhitelisted: true` or even without that?